hack/generated/pkg/genruntime/base_types.go

```diff
@@ -15,6 +15,15 @@ import (
 	"github.com/Azure/azure-service-operator/hack/generated/pkg/genruntime/conditions"
 )
 
+type ResourceKind string
+
+const (
+	// ResourceKindNormal is a standard ARM resource.
+	ResourceKindNormal = ResourceKind("normal")
+	// ResourceKindExtension is an extension resource. Extension resources can have any ResourceKindNormal resource as their parent.
```

Comment:
I don't think so, but let me confirm...

Reply:
Actually, it doesn't matter. Even if it's not allowed there's nothing stopping us from representing it. I'll correct this documentation to clarify that it can be _any_ resource (rather than just any Normal resource).